### PR TITLE
community/vault: upgrade to 1.2.3 and improve init script

### PR DESCRIPTION
**Pull request by mtneug**
Upgrade Vault to latest 1.2.3 and improve init scrip with supervise-daemon configurations and config `reload` functions.

### CODE REVIEW

**Comment by maxice8** [comment]
upgrade is lowercase

File: community/vault/vault.initd [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
-command=/usr/sbin/${RC_SVCNAME}
+command="/usr/sbin/${RC_SVCNAME}"
 command_args="${vault_opts}"
 command_user="${RC_SVCNAME}:${RC_SVCNAME}"
 
-start_pre() {
-	checkpath -f -m 0644 -o "$command_user" "$VAULT_LOG_FILE"
-}
+supervisor=supervise-daemon
+pidfile="/run/${RC_SVCNAME}.pid"
```
Comment (maxice8) [inline community/vault/vault.initd]: don't think pidfiles matter for supervise-daemon

File: community/vault/vault.initd [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
 }
+
+start_pre() {
+	checkpath -f -m 0644 -o "$command_user" "$output_log" "$error_log"
+}
+
+reload() {
+	start_pre \
+		&& ebegin "Reloading $RC_SVCNAME configuration" \
+		&& supervise-daemon "$RC_SVCNAME" --signal HUP --pidfile "$pidfile"
```
Comment (maxice8) [inline community/vault/vault.initd]: use $supervisor, supervise-daemon shouldn't use pidfile

File: community/vault/vault.initd (hunk 1 above)
Comment (mtneug) [reply community/vault/vault.initd]: Apparently supervise-daemon will create a default PID file (`/run/supervise-vault.pid` in this case). Will update the PR.

**Pull request closed without merging**